pyquil/latex/_diagram.py

```diff
@@ -506,6 +505,9 @@ def _build_generic_unitary(self) -> None:
             offset = target_qubits[0] - q
             self.diagram.append(q, TIKZ_CONTROL(q, offset))
 
+        # sort the qubit list because the first qubit indicates wire placement on the diagram
+        target_qubits.sort()
+
```

Comment:
`target_qubits` gets sorted on l501 and l509, but is it safe to just sort it once on l500? We'd have to check if that introduces a new bug in l504-506, which is operating on the unsorted array. To me it looks like it's just calculating the offset to put in the `\ctrl{offset}` latex statement, which I think we should be able to accommodate, even with a sorted array. What do you think? Could we text it by adding multiple control modifiers to any basic gate (ie. `X`)?

Reply:
I moved the sorting to where `target_qubits` is set, `target_qubits = sorted(qubits[controls:])`, and ran the test_latex.py module and then the entire test suite (using `--runslow` on all of the tests that timed out) and everything works! From my understanding, this is ok because we're simply sorting the target qubits. If the control qubits were included in the sorting then maybe we'd run into problems. 

I tried producing the LaTeX circuit for `p += X(0).controlled(1).controlled(2).controlled(3)`

<img width="508" alt="Screenshot 2023-01-23 at 8 39 36 PM" src="https://user-images.githubusercontent.com/58452495/214213633-9c39bf12-8fdf-46f3-8394-4d0d7b0b15c6.png">

I also tried `p += X(3).controlled(2).controlled(1).controlled(0)` 

<img width="508" alt="Screenshot 2023-01-23 at 8 45 26 PM" src="https://user-images.githubusercontent.com/58452495/214213984-1e2145df-3675-45f8-9d80-1e65b988a3fb.png">


And finally `p += X(2).controlled(3).controlled(0).controlled(1)` 

<img width="511" alt="Screenshot 2023-01-23 at 8 44 21 PM" src="https://user-images.githubusercontent.com/58452495/214213873-436d2b45-4b9b-4382-abd4-88c44d2d49d9.png">

They all seem to be generating all the expected circuits. Really there's only 1 target qubit though, so sorting it isn't really doing anything 😆